tests/unit/components/UserText/UserText.test.js

```diff
@@ -0,0 +1,154 @@
+  } );
+
+  it( "links all URLs", () => {
+    const testText = "https://www.inaturalist.org";
+    const { getByRole, queryByText } = render(
+      <UserText text={testText} />
+    );
+
+    expect( getByRole( "link" ) ).toBeTruthy();
+    expect( queryByText( testText ) ).toBeTruthy();
+  } );
+
+  it( "closes unclosed tags", () => {
+    const testText = "<p>Welcome to iNat";
+    const { queryByText } = render(
+      <UserText text={testText} />
+    );
+
+    expect( queryByText( "Welcome to iNat" ) ).toBeTruthy();
+  } );
+
+  it( "strips leading and trailing whitespace", () => {
+    const testText = " This is a single line with a lloooooot of whitespace   \n\n\n\n\n\n\n      ";
+    const { queryByText } = render(
+      <UserText text={testText} />
+    );
+
+    expect( queryByText( "This is a single line with a lloooooot of whitespace" ) ).toBeTruthy();
+    expect( queryByText( testText ) ).toBeFalsy();
+  } );
+} );
+
+describe( "Basic Rendering", () => {
+  it( "renders text", () => {
+    const testText = "foo bar baz";
+    const { queryByText, getByText } = render(
+      <UserText text={testText} />
+    );
+
+    expect( getByText( testText ) ).toBeTruthy();
+    expect( queryByText( "asdfgh" ) ).toBeFalsy();
+  } );
+
+  it( "renders markdown", () => {
+    const testText = "# This is Heading 1";
+    const { queryByText } = render(
+      <UserText text={testText} />
+    );
+
+    expect( queryByText( testText ) ).toBeFalsy();
+    expect( queryByText( "This is Heading 1" ) ).toBeTruthy();
+    // eslint-disable-next-line max-len
+    expect( queryByText( "This is Heading 1" ) ).toHaveProperty( "props.style.0.fontWeight", "bold" );
+  } );
+
+  it( "renders html", () => {
+    const testText = "<p>Welcome to <b>iNaturalist</b></p>";
+    const { queryByText } = render(
+      <UserText text={testText} />
+    );
+
+    expect( queryByText( testText ) ).toBeFalsy();
+    expect( queryByText( "Welcome to" ) ).toBeTruthy();
+    expect( queryByText( "iNaturalist" ) ).toBeTruthy();
+    expect( queryByText( "Welcome to" ) ).not.toHaveProperty( "props.style.0.fontWeight", "bold" );
+    expect( queryByText( "iNaturalist" ) ).toHaveProperty( "props.style.0.fontWeight", "bold" );
+  } );
+
+  it( "Parse Markdown, tables", () => {
+    const testText = ( `| # | Name   | Age
+      |---|--------|-----|
+      | 1 | John   | 19  |
+      | 2 | Sally  | 18  |
+      | 3 | Stream | 20  |` );
+    const { queryByText } = render(
+      <UserText text={testText} />
+    );
+
+    expect( queryByText( testText ) ).toBeFalsy();
+    expect( queryByText( "| # | Name | Age" ) ).toBeTruthy();
+  } );
+
+  it( "Parse Markdown, lists", () => {
+    const testText = ( `
+    1. List1
+    2. List2
+    3. List3
+    4. List4.` );
+    const { queryByText } = render(
+      <UserText text={testText} />
+    );
+    expect( queryByText( testText ) ).toBeFalsy();
```

Comment:
What can we do to ensure a list is getting rendered and not just ensuring that the markup itself is *not* getting rendered?